### DISCUSSION TITLE
No local development option for skipping nonce checks for iOS Google Sign In

### DISCUSSION THREAD

**Discussion by jgfino** [supabase/supabase#33057] [Troubleshooting]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

The issue outlined in https://github.com/supabase/supabase/issues/17999 still occurs for those of us using Supabase to develop locally

## To Reproduce

Follow the [Expo Docs](https://docs.expo.dev/guides/google-authentication/) for Google sign in and connect your app to a **local** Supabase environment

## Expected behavior

Google sign in should work correctly without the `[AuthApiError: Passed nonce and nonce in id_token should either both exist or not.]` error by being able to disable nonce checking

It works flawlessly when connected to an actual hosted Supabase project with the nonce check disabled, but I still get the error when trying to test Google sign in locally. 

From what I can see, there is no option for local development to disable the nonce check (similar to what was done in https://github.com/supabase/supabase/pull/18301)

## Additional context

Is there currently a `config.toml` option for this? or could one be added? I might take a look myself if I have time but wanted to see if anyone else had a solution to this or has also run into this problem

**Comment by joshsmith** [comment]
[According to the docs](https://supabase.com/docs/guides/cli/config) there is nothing in the `auth.external.provider` configs that provide for this option. Also would love to see a config option for this.

**Comment by devjiro76** [comment]
I have same problem. I tried to use google-login on iOS simulator.
Same error [AuthApiError: Passed nonce and nonce in id_token should either both exist or not.] is occured.
It will be very happy to find the option for skip nonce.

**Comment by erickreutz** [comment]
We really need this. I don't know how we are expected to even use `supabase start` if we cant configure it to mimic supabase.com

**Comment by Arjit222830** [comment]
Same Issue

**Comment by rahitecho** [comment]
Same issue

**Comment by dhutaryan** [comment]
Same issue

**Comment by albytterc** [comment]
Same issue

**Comment by peter-lee97** [comment]
I've updated to supabase version 1.163.2 and the issue is still persisting.

**Comment by Heidelbergg** [comment]
Any workarounds?

**Comment by shakthizen** [comment]
Same issue

**Comment by shakthizen** [comment]
I found out that in gotru you can use this env variable to disable nonce check

 ```
GOTRUE_EXTERNAL_GOOGLE_SKIP_NONCE_CHECK=false
```

It is not documented in the readme file but implemented in the code.

**Comment by cartellink** [comment]
This worked for me.

```
[auth.external.google]
enabled = true
client_id = ""
secret = ""
# Overrides the default auth redirectUrl.
redirect_uri = "http://localhost:54321/auth/v1/callback"
skip_nonce_check = true
url = ""
```

**Comment by dhutaryan** [comment]
> This worked for me.
> 
> ```
> [auth.external.google]
> enabled = true
> client_id = ""
> secret = ""
> # Overrides the default auth redirectUrl.
> redirect_uri = "http://localhost:54321/auth/v1/callback"
> skip_nonce_check = true
> url = ""
> ```

I updated supabase and it works for me as well.

**Comment by zan-12** [comment]
> I found out that in gotru you can use this env variable to disable nonce check
> 
> ```
> GOTRUE_EXTERNAL_GOOGLE_SKIP_NONCE_CHECK=false
> ```
> 
> It is not documented in the readme file but implemented in the code.



I had the same issue, and it was doing the nonce check with that value, but setting it to `true` worked for me!

  ```
  GOTRUE_EXTERNAL_GOOGLE_SKIP_NONCE_CHECK=true
  ```

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements for reference troubleshooting.
